src/Umbraco.Infrastructure/Cache/DefaultRepositoryCachePolicy.cs

```diff
@@ -137,7 +139,7 @@ public override void Delete(TEntity entity, Action<TEntity> persistDeleted)
 
         TEntity? entity = performGet(id);
 
-        if (entity != null && entity.HasIdentity)
+        if ((entity != null && entity.HasIdentity) || _options.CacheNullValues)
```

Comment:
There might not be an existing situation, but this check will potentially allow caching non-null entities that have no identity, are we OK with that @Migaroez ?